pkg/controller/wildflyserver/wildflyserver_controller.go

```diff
@@ -504,20 +550,339 @@ func (r *ReconcileWildFlyServer) routeForWildFly(w *wildflyv1alpha1.WildFlyServe
+	scaleDownPodName := scaleDownPod.ObjectMeta.Name
+	scaleDownPodIP := scaleDownPod.Status.PodIP
+	jbossHome := getJbossHome()
+
+	// Reading timestamp for the latest log record
+	scaleDownPodLogTimestampAtStart, err := wildflyutil.ObtainLogLatestTimestamp(scaleDownPod)
+	if err != nil {
+		return false, "", fmt.Errorf("Failed to read log from scaling down pod '%v', error: %v", scaleDownPodName, err)
+	}
+
+	// TODO: consider if we can get rid of the recovery listener enablement to speed up the scale down processing
+	// Enabling recovery listener to speed up recovery processing
+	jsonResult, err := wildflyutil.ExecuteMgmtOp(scaleDownPod, jbossHome, wildflyutil.MgmtOpTxnEnableRecoveryListener)
+	if err != nil {
+		return false, "", fmt.Errorf("Cannot enable transaction recovery listener for scaling down pod %v, error: %v", scaleDownPodName, err)
+	}
+	if !wildflyutil.IsMgmtOutcomeSuccesful(jsonResult) {
+		return false, "", fmt.Errorf("Failed to enable transaction recovery listener for scaling down pod %v. Scaledown processing cannot trigger recovery. "+
+			"Management command: %v, JSON response: %v", scaleDownPodName, wildflyutil.MgmtOpTxnEnableRecoveryListener, jsonResult)
+	}
+	// Enabling the recovery listner may require the server being reloaded
+	isReloadRequired := wildflyutil.ReadJSONDataByIndex(jsonResult["response-headers"], "operation-requires-reload")
+	if isReloadRequiredBool, _ := strconv.ParseBool(isReloadRequired); isReloadRequiredBool {
+		jsonResult, err = wildflyutil.ExecuteMgmtOp(scaleDownPod, jbossHome, wildflyutil.MgmtOpReload)
+		if err != nil {
+			return false, "", fmt.Errorf("Cannot reload application container for scaling down pod %v, error: %v", scaleDownPodName, err)
+		}
+		for serverStateCheckCounter := 0; err != nil && serverStateCheckCounter < reloadRetryCount; serverStateCheckCounter++ {
+			jsonResult, err = wildflyutil.ExecuteMgmtOp(scaleDownPod, jbossHome, wildflyutil.MgmtOpServerStateRead)
+		}
+		if err != nil { // reload did not yet finished and server is not properly running
+			return false, "", fmt.Errorf("Failed waiting for server to be reloaded for the scaling down pod %v, error: %v", scaleDownPodName, err)
+		}
+	}
+	// Reading recovery port from the app server with management port
+	scaleDownPodRecoveryPort, err := wildflyutil.GetTransactionRecoveryPort(scaleDownPod, jbossHome)
+	if err != nil || scaleDownPodRecoveryPort == 0 {
+		scaleDownPodRecoveryPort = defaultRecoveryPort
+	}
+	if err != nil {
+		reqLogger.Error(err, "Error on reading transaction recovery port with management command")
+	}
+	// With enabled recovery listener and the port, let's start the recovery scan
+	reqLogger.Info("Executing recovery scan", "Pod name", scaleDownPodName, "Pod IP", scaleDownPodIP, "Recovery port", scaleDownPodRecoveryPort)
+	_, err = wildflyutil.SocketConnect(scaleDownPodIP, scaleDownPodRecoveryPort, txnRecoveryScanCommand)
```

Comment:
out of curiosity, why is the recovery scan not exposed as a management operation in the transaction subsystem?

Reply:
As this was not my design decision I'm not 100% sure about that. But in general I can say that forcing the recovery is not a standard operation that is about to be offered to a usual user. It's internal Narayana procedure which is not expected to be touched. It's helpful only for a person who really understands how Narayana works and what happens when it's called. It's useful for testing and for integration purposes. But I can't imagine that would be useful for an ordinary developer to force the recovery processing.
What a developer/admin/an op person can change is the period for the processing. This is currently possible only via system properties (not available in model).

As I wonder what are all the reasons behind  I created a forum question about this at: https://developer.jboss.org/message/990677#990677